Mlem/Models/Content/Community/Community List/CommunityListModel.swift

```diff
@@ -205,28 +119,125 @@ class CommunityListModel: ObservableObject {
+            CommunityListSection(
+                viewId: "favorites",
+                sidebarEntry: FavoritesSidebarEntry(
+                    sidebarLabel: nil,
+                    sidebarIcon: "star.fill"
+                ),
+                inlineHeaderLabel: "Favorites",
+                accessibilityLabel: "Favorited Communities",
+                communities: favorited
+            )
+        }
+        newAllSections.append(favoritesSection)
+        if !favoritedSet.isEmpty {
+            newVisibleSections.append(favoritesSection)
+        }
         
-        // update our published value for the view to render
-        mainQueue.schedule { [weak self] in
-            self?.communities = sorted
+        let alphabeticSections = alphabeticSections()
+        newAllSections.append(contentsOf: alphabeticSections)
+        newVisibleSections.append(contentsOf: alphabeticSections.filter { section in
+            !section.communities.isEmpty
+        })
+        
+        let nonLetterSections = withDependencies(from: self) {
+            let sidebarEntry = RegexCommunityNameSidebarEntry(
+                communityNameRegex: /^[^a-zA-Z]/,
+                sidebarLabel: "#",
+                sidebarIcon: nil
+            )
+            
+            return CommunityListSection(
+                viewId: "non_letter_titles",
+                sidebarEntry: sidebarEntry,
+                inlineHeaderLabel: "#",
+                accessibilityLabel: "Communities starting with a symbol or number",
+                communities: allCommunities.filter { community in
+                    sidebarEntry.contains(community: community, isSubscribed: subscribedSet.contains(community.id))
+                }
+            )
+        }
+        newAllSections.append(nonLetterSections)
+        if !nonLetterSections.communities.isEmpty {
+            newVisibleSections.append(nonLetterSections)
+        }
+        
+        return (all: newAllSections, visible: newVisibleSections)
+    }
+    
+    private func alphabeticSections() -> [CommunityListSection] {
+        let alphabet: [String] = .alphabet
+        return alphabet.map { character in
+            withDependencies(from: self) {
+                let sidebarEntry = RegexCommunityNameSidebarEntry(
+                    communityNameRegex: (try? Regex("^[\(character.uppercased())\(character.lowercased())]"))!,
+                    sidebarLabel: character,
+                    sidebarIcon: nil
+                )
+                
+                // This looks sinister but I didn't know how to string replace in a non-string based regex
+                return CommunityListSection(
+                    viewId: character,
+                    sidebarEntry: sidebarEntry,
+                    inlineHeaderLabel: character,
+                    accessibilityLabel: "Communities starting with the letter '\(character)'",
+                    communities: allCommunities.filter { community in
```

Comment:
Because the list is already sorted alphabetically, there's probably a better alternative to using `filter` here. It doesn't change the time complexity though, so not a huge deal

Reply:
Optimized it anyway--*26n* might be the same as *n* for algo nerds, but a factor of 26 is significant in my book